volmdlr/wires.py

```diff
@@ -1445,6 +1450,43 @@ def triangulation(self):
                           initial_point_to_index[p3]))
         
         return vmd.DisplayMesh2D(points, triangles)
+    def lenght(self):
+        L = []
+        for k in range(len(self.line_segments)):
+            L.append(self.line_segments[k].length())
+        return sum(L)
+    def update_polygon(self, points):
+        self.points = points
+        self.line_segments = self._line_segments()
+    def statistics(self):
+        distances=[]
+        for i, point in enumerate(self.points):
+            if i!=0:
+                distances.append(point.point_distance(self.points[i-1]))
+        mean_distance = mean(distances)
+        std = npy.std(distances)
+        return mean_distance, std
+                
+    def simplify_polygon2d(self):
+        points = [self.points[0]]
+        mean_distance, _ = self.statistics()
+        for i, point in enumerate(self.points):
+            if i != 0:
+                distance = point.point_distance(points[-1])
+                if distance > mean_distance:
+                    # if distance > 10*mean_distance:
+                    #     mean_point = 0.5*(point+points[-1])
+                    #     mean_point2 = 0.5*(mean_point+points[-1])
+                    #     mean_point3 = 0.5*(point+mean_point)
+                    #     points.append(mean_point2)
+                    #     points.append(mean_point)
+                    #     points.append(mean_point3)
+                   
+                    points.append(point)
+               
+        self.update_polygon(points)
+        
```

Comment:
Put in ClosedPolygon2D and 3D: Wire does not have self.points!

Reply:
What do you mean? So I can't update the polygon points like this?